src/kaleidoscope/driver/storage/AVREEPROM.h

```diff
@@ -0,0 +1,59 @@
+/* -*- mode: c++ -*-
+ * kaleidoscope::driver::storage::AVREEPROM -- Storage driver with AVR EEPROM backend
+ * Copyright (C) 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "kaleidoscope/driver/storage/Base.h"
+#include <EEPROM.h>
+
+namespace kaleidoscope {
+namespace driver {
+namespace storage {
+
+struct AVREEPROMProps : kaleidoscope::driver::storage::BaseProps {
```

Comment:
One of the things most likely to change between different AVR MCUs is how much eeprom they have. This might want to be ATMega32U4EEPROM? 

Or we could have ATMega32U4EEPROM extend AVREEPROM.

Reply:
I don't really see the point of having both `AVREEPROMProps` and `ATMega32U4EEPROMProps`, since - at least as far as I remember - the only difference in props would be the length. I'll rename to `ATMega32U4EEPROMProps`.

Reply:
...I don't want to swear to all AVR eeproms functioning identically. As a general pattern, if we're going to have implementations for multiple MCUs in a family, it seems like it'd be a clean design to have them all inherit from the same place.

THOUGH. It might be worth seeing if avr-gcc offers a standard definition for how much storage the currently configured microcontroller has. Might be. and might just be worth opening a ticket or a // TODO for later experimentation.